### ISSUE TITLE
Add CDF method to Bucketizer class

### ISSUE THREAD

**Issue by afiq-aswadi** [afiq-aswadi/pfn_transformers#2]
## Summary

Add a `cdf()` method to the `Bucketizer` class that computes the cumulative distribution function at arbitrary continuous values.

## Motivation

Currently `Bucketizer` has:
- `sample()` - draw from the distribution  
- `log_density_at_values()` - evaluate log PDF at points
- `log_bucket_densities()` - get log densities per bucket

A `cdf()` method would complete the interface and enable:
- Martingale property testing for PFN predictive distributions
- Probability calibration checks
- PP-plots and QQ-plots for distribution comparison

## Proposed API

```python
def cdf(
    self,
    logits: Float[Tensor, "... num_buckets"],
    values: Float[Tensor, "..."],
) -> Float[Tensor, "..."]:
    """Evaluate CDF F(y) at arbitrary continuous values.
    
    For bounded support: piecewise-linear interpolation (uniform within buckets)
    For unbounded support: half-normal CDFs for edge buckets, linear for interior
    
    Args:
        logits: Logits over buckets
        values: Points at which to evaluate the CDF
        
    Returns:
        CDF values F(y) in [0, 1]
    """
```

## Implementation Notes

For **bounded support**:
- Interior buckets: `F(y) = Σ_{j<i} p_j + p_i × (y - b_i) / width_i`

For **unbounded support**:
- Left edge bucket: `F(y) = p_0 × (1 - erf((borders[1] - y) / (scale × √2)))`
- Right edge bucket: `F(y) = mass_before + p_{n-1} × erf((y - borders[-2]) / (scale × √2))`
- Interior buckets: same as bounded

The half-normal scale uses: `scale = width / HALF_NORMAL_ICDF_AT_TARGET` (same as existing code in `bucketizer.py`).

## Implementation Requirements

- Vectorized (no Python loops)
- GPU-compatible  
- Follow the same patterns as `log_density_at_values()`

**Comment by afiq-aswadi** [comment]
## Implementation Plan (Local Checkpoint Loading)

Moving away from wandb artifact loading in favor of local checkpoints.

### Key Update
The `pfn_transformers` submodule now saves task distributions in checkpoints:
- `train.py` extracts `prior.tasks` and saves as `task_distribution` dict
- `load_checkpoint()` attaches it to model as `model.task_distribution`

New training runs will have task pools embedded. Existing checkpoints won't (will need re-training).

### Proposed Implementation

1. **YAML config** (`configs/sweep_runs.yaml`) - maps run_id to num_tasks
2. **Helper script** (`scripts/generate_sweep_config.py`) - auto-generate YAML from checkpoint directories  
3. **Add Wasserstein** to `baselines.py` - sliced Wasserstein via random projections
4. **Main analysis script** (`experiments/exp_sweep_analysis.py`):
   - Tyro CLI with ExpConfig
   - Load models via `load_checkpoint()`, extract task pool from `model.task_distribution["tasks"]`
   - Compute predictive metrics: MSE, delta_ridge, delta_dmmse
   - Compute prior metrics: energy_distance, sliced_wasserstein
   - Output: CSV + plots (log-scale x-axis for num_tasks)

### Metrics
| Metric | Space |
|--------|-------|
| MSE | Predictive |
| Delta Ridge | Predictive |
| Delta dMMSE | Predictive |
| Energy distance (PT vs dMMSE/ridge) | Prior |
| Sliced Wasserstein (PT vs dMMSE/ridge) | Prior |

### Next Steps
- [ ] Run num_tasks sweep with updated submodule (checkpoints will include task pools)
- [ ] Implement analysis script
